www/src/pages/table/forms/TransformationForm.tsx

```diff
@@ -0,0 +1,289 @@
+                >
+                  Execute
+                </Button>
+              </Form.Item>
+            </Col>
+            <Col span={2} flex="auto">
+              <Form.Item>
+                <Button style={{ width: "100%" }} type="primary">
+                  Save
+                </Button>
+              </Form.Item>
+            </Col>
+            <Col span={2} flex="auto">
+              <Form.Item>
+                <Button style={{ width: "100%" }} type="primary">
+                  Reset
+                </Button>
+              </Form.Item>
+            </Col>
+            <Col span={11} style={{ paddingLeft: "2rem" }}>
+              <Form.Item
+                name="onerror"
+                label="on error"
+                rules={[{ required: true }]}
+                initialValue={4}
+              >
+                <Radio.Group>
+                  <Radio value={1}>Set To Blank</Radio>
+                  <Radio value={2}>Store Error</Radio>
+                  <Radio value={3}>Keep Original</Radio>
+                  <Radio value={4}>Abort</Radio>
+                </Radio.Group>
+              </Form.Item>
+            </Col>
+            <Space></Space>
+            <Col span={4}>
+              <Form.Item
+                name="tolerance"
+                label="tolerance"
+                rules={[{ required: true }]}
+                initialValue={3}
+              >
+                <InputNumber placeholder="3" />
+              </Form.Item>
+            </Col>
+            <Col span={3}>
+              <Form.Item name="rows" label="rows">
+                <InputNumber />
+              </Form.Item>
+            </Col>
+          </Row>
+          <Row>
+            <Col span={24}>
+              {Array.isArray(result) ? (
+                <Table
+                  style={{ border: "1px solid #ccc" }}
+                  columns={columns}
+                  dataSource={result.map(transformationResult2Row)}
+                  pagination={{ pageSize: 4 }}
+                />
+              ) : result != undefined ? (
+                <Tag style={{ padding: 0 }} color={"volcano"}>
+                  <pre style={{ margin: 5 }}>{result}</pre>
+                </Tag>
+              ) : (
+                <></>
+              )}
+            </Col>
+          </Row>
+        </Form>
+      );
+    }
+  )
+);
+
+export const CustomEditor = ({
+  value,
+  onChange,
+}: {
+  value?: string;
+  onChange?: (value: string | undefined) => void;
+}) => {
+  const options = {
+    autoIndent: "full",
+    contextmenu: true,
+    fontFamily: "monospace",
+    fontSize: 13,
+    lineHeight: 24,
+    matchBrackets: "always",
+    minimap: {
+      enabled: false,
+    },
+    scrollbar: {
+      horizontalSliderSize: 4,
+      verticalSliderSize: 18,
+    },
+    selectOnLineNumbers: true,
+    roundedSelection: false,
+    readOnly: false,
+    cursorStyle: "line",
+    automaticLayout: true,
+  };
```

Comment:
this doesn't change everytime we re-render the component, it's better to move it outside.

Reply:
updated it.